fuzzers/jif/libjif/src/bin/libafl_cc.rs

```diff
@@ -0,0 +1,431 @@
+//! Uses LLVM compiler Wrapper from `LibAFL`
+
+use libafl_cc::{
+    clang::{CLANGXX_PATH, CLANG_PATH, OUT_DIR},
+    CompilerWrapper, Error, LLVMPasses, LIB_EXT, LIB_PREFIX,
+};
+
+use std::{
+    convert::Into,
+    env,
+    path::{Path, PathBuf},
+    string::String,
+    vec::Vec,
+};
+
+/// Wrap Clang
+#[allow(clippy::struct_excessive_bools)]
+#[derive(Debug)]
+pub struct ClangWrapper {
+    is_silent: bool,
+    optimize: bool,
+    wrapped_cc: String,
+    wrapped_cxx: String,
+
+    name: String,
+    is_cpp: bool,
+    linking: bool,
+    x_set: bool,
+    bit_mode: u32,
+    need_libafl_arg: bool,
+    has_libafl_arg: bool,
+
+    parse_args_called: bool,
+    base_args: Vec<String>,
+    cc_args: Vec<String>,
+    link_args: Vec<String>,
+    passes: Vec<LLVMPasses>,
+    passes_args: Vec<String>,
+}
+
+#[allow(clippy::match_same_arms)] // for the linking = false wip for "shared"
+#[allow(clippy::too_many_lines)]
+#[allow(clippy::case_sensitive_file_extension_comparisons)]
+impl CompilerWrapper for ClangWrapper {
```

Comment:
@andreafioraldi that's where we needed `OUT_DIR` for, but probably there is a better way to reuse the libafl_cc `ClangWrapper` and have some custom callbacks in there

Reply:
@jhertz can you describe the usecase?

Reply:
"Make Chrome link to LibAFL"

Reply:
chrome requires a specific version of clang that it vendors in. we are using clang-13 here for reasons.

so we have to ignore certain flags, and add certain flags, to make everything link. this is totally custom and "per project"